lib/pg/logical_replication/client.rb

```diff
@@ -59,6 +59,9 @@ def wal_retained_bytes
       # @param publications [Array<String>] publication names to subscribe to
       # @param options [Hash] optional parameters for CREATE SUBSCRIPTION
       def create_subscription(name, conninfo_hash, publications, options = {})
+        options = options.transform_keys(&:to_s)
+        options['slot_name'] ||= name if options['create_slot'] == false
```

Comment:
I'm guess what I'm saying is we could avoid the Ruby version break by just doing:

```suggestion
        options[:slot_name] = name if !options.key?(:slot_name) && !options.key?("slot_name") && (options['create_slot'] == false || options[:create_slot] == false)
```

Reply:
yeah, I can do this.  I had something like this originally.